flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-paimon/src/test/java/org/apache/flink/cdc/connectors/paimon/sink/PaimonMetadataApplierTest.java

```diff
@@ -441,6 +441,13 @@ public void testAddColumnWithPosition(String metastore)
                 tableSchema, catalog.getTable(Identifier.fromString("test.table1")).rowType());
 
         addedColumns.clear();
+
+        addedColumns.add(
+                AddColumnEvent.before(
+                        Column.physicalColumn(
+                                "col4_first_before",
+                                org.apache.flink.cdc.common.types.DataTypes.STRING()),
+                        "col1"));
```

Comment:
So the before column is actually deduced by SchemaRegistry?

Reply:
The before column is actually derived in [schemaUtils](https://github.com/apache/flink-cdc/blob/e7964b160c1df3861571ba2199ba04d91d887ff2/flink-cdc-common/src/main/java/org/apache/flink/cdc/common/utils/SchemaUtils.java#L239). If the transform is defined in the pipeline
```
transform:
- source-table: dw_app.test_table
projection: __table_name__ as table_name,\*
primary-keys: id
```

At this time, when add column first is executed in the mysql table, the [add column first] event will become the [add column before] event.
If no transform is defined, add first first event remains the same